examples/chip-tool/commands/interactive/InteractiveCommands.cpp

```diff
@@ -65,6 +66,132 @@ char * GetCommand(char * command)
     return command;
 }
 
+void LogWebSocketCallbackReason(lws_callback_reasons reason)
+{
+#if CHIP_DETAIL_LOGGING
+    switch (reason)
+    {
+    case LWS_CALLBACK_GET_THREAD_ID:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_GET_THREAD_ID");
+        break;
+    case LWS_CALLBACK_ADD_HEADERS:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_ADD_HEADERS");
+        break;
+    case LWS_CALLBACK_PROTOCOL_INIT:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_PROTOCOL_INIT");
+        break;
+    case LWS_CALLBACK_PROTOCOL_DESTROY:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_PROTOCOL_DESTROY");
+        break;
+    case LWS_CALLBACK_HTTP:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_HTTP");
+        break;
+    case LWS_CALLBACK_EVENT_WAIT_CANCELLED:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_EVENT_WAIT_CANCELLED");
+        break;
+    case LWS_CALLBACK_CLIENT_WRITEABLE:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_CLIENT_WRITEABLE");
+        break;
+    case LWS_CALLBACK_FILTER_NETWORK_CONNECTION:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_FILTER_NETWORK_CONNECTION");
+        break;
+    case LWS_CALLBACK_FILTER_PROTOCOL_CONNECTION:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_FILTER_PROTOCOL_CONNECTION");
+        break;
+    case LWS_CALLBACK_WSI_CREATE:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_WSI_CREATE");
+        break;
+    case LWS_CALLBACK_SERVER_NEW_CLIENT_INSTANTIATED:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_SERVER_NEW_CLIENT_INSTANTIATED");
+        break;
+    case LWS_CALLBACK_HTTP_CONFIRM_UPGRADE:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_HTTP_CONFIRM_UPGRADE");
+        break;
+    case LWS_CALLBACK_HTTP_BIND_PROTOCOL:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_HTTP_BIND_PROTOCOL");
+        break;
+    case LWS_CALLBACK_ESTABLISHED:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_ESTABLISHED");
+        break;
+    case LWS_CALLBACK_RECEIVE:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_RECEIVE");
+        break;
+    case LWS_CALLBACK_WS_PEER_INITIATED_CLOSE:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_WS_PEER_INITIATED_CLOSE");
+        break;
+    case LWS_CALLBACK_WSI_DESTROY:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_WSI_DESTROY");
+        break;
+    case LWS_CALLBACK_CLOSED:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_CLOSED");
+        break;
+    case LWS_CALLBACK_SERVER_WRITEABLE:
+        ChipLogDetail(chipTool, "LWS_CALLBACK_SERVER_WRITEABLE");
+        break;
+    default:
+        ChipLogError(chipTool, "Unknown reason: %d ", static_cast<int>(reason));
+    }
+#endif // CHIP_DETAIL_LOGGING
+}
+
+static int OnWebSocketCallback(lws * wsi, lws_callback_reasons reason, void * user, void * in, size_t len)
+{
+    LogWebSocketCallbackReason(reason);
+
+    if (LWS_CALLBACK_RECEIVE == reason)
+    {
+        char * command = static_cast<char *>(chip::Platform::MemoryCalloc(len + 1 /* for null byte */, sizeof(char)));
```

Comment:
Please add a comment on how this is freed. Since this is not clearly scoped, the lifetime of this is not obvious. Maybe also explanation that ParseCommand may point inside this may be useful in understanding why this is allocated like this.